src/vcf/validator.cpp

```diff
@@ -132,6 +132,65 @@ namespace ebi
         }
     }
 
+    bool is_compressed_file(std::istream &input,
+                            const std::string &source)
+    {
+        std::string message = "Input file is a compressed file ";
+
+        if (source != ebi::vcf::STDIN) {
+            boost::filesystem::path source_name(source);
+            std::string file_extension = source_name.extension().string();
+
+            if (file_extension == ".zip") {
```

Comment:
You could extract these strings ("zip", "tar", etc) to the file `string_constants.hpp`.

Also the multiple `if` statements could be merged into one single statement.

Reply:
sure.
I will move the strings to `inc/vcf/string_constants.hpp` file.
Should I move the warning message strings too ?
I was checking in `validator_main.cpp` few strings(used in info messages) were used there only.

Reply:
we are not extracting messages, only words that can actually provoke bugs if misspelled. extracting the messages only makes sense for i18n, which we are not likely to do in the short to mid term